omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,346 @@
+			           " \"reaction\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\" }" +
+			         "}," +
+			         "{" +
+					   " \"allergy\" : { \"uuid\" : \"" + PropertyUtils.getProperty(so, "uuid") + "\"}," +
+			           " \"reaction\" : { \"uuid\" : \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\" }," +
+			           " \"reactionNonCoded\" : \"test non coded reaction\"" +
+			         "}" +
+			       "]" +
+		       "}";
+		so = deserialize(handle(newPostRequest(getURI() + "/" + PropertyUtils.getProperty(so, "uuid"), json)));
+
+		Assert.assertEquals("allergy comment", Util.getByPath(so, "comment"));
+		Assert.assertEquals("35d3346a-6769-4d52-823f-b4b234bac3e3", Util.getByPath(so, "severity/uuid"));
+
+		Assert.assertEquals("35d3346a-6769-4d52-823f-b4b234bac3e3", Util.getByPath(so, "allergen/codedAllergen/uuid"));
+		Assert.assertEquals("DRUG", Util.getByPath(so, "allergen/allergenType"));
+		
+		Assert.assertEquals("35d3346a-6769-4d52-823f-b4b234bac3e3", Util.getByPath(so, "reactions[0]/reaction/uuid"));
+
+		Assert.assertEquals("5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA", Util.getByPath(so, "reactions[1]/reaction/uuid"));
+		Assert.assertEquals("test non coded reaction", Util.getByPath(so, "reactions[1]/reactionNonCoded"));
+		
+		// assert that a new allergy has been added
+		allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 5);
+	}
+	
+	/**
+	 *  Save Allergy with Non Coded Allergen
+	 */
+	@Test
+	public void shouldSaveAllergyWithNonCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 4);
+		
+		// save allergy with non coded allergen
+		String json = "{" +
+				        " \"comment\" : \"allergy comment\"," +
+				        " \"severity\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\"}," +
+				        " \"allergen\" : " +
+				          " { \"allergenType\" : \"DRUG\", " +
+				          " \"codedAllergen\" : { \"uuid\" : \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\"}, " +
+				          " \"nonCodedAllergen\" : \"test non coded allergen\"} " +
+		              "}";
+		
+		// save allergy with reactions to get uuid for later saving of reactions
+		SimpleObject so = deserialize(handle(newPostRequest(getURI(), json)));
+		
+		// update allergy with reactions
+		json = "{ \"reactions\" : " +
+			       "[" +
+			         "{" +
+					   " \"allergy\" : { \"uuid\" : \"" + PropertyUtils.getProperty(so, "uuid") + "\"}," +
+			           " \"reaction\" : { \"uuid\" : \"35d3346a-6769-4d52-823f-b4b234bac3e3\" }" +
+			         "}," +
+			         "{" +
+					   " \"allergy\" : { \"uuid\" : \"" + PropertyUtils.getProperty(so, "uuid") + "\"}," +
+			           " \"reaction\" : { \"uuid\" : \"5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\" }," +

```

Comment:
I thought you were going to create a separate test method for reactions and give it a name that contains "shouldSaveAllergyWithReactions"